compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -329,20 +331,42 @@ object JavaParsers {
     }
 
     def annotations(): List[Tree] = {
-      //var annots = new ListBuffer[Tree]
+      var annots = new ListBuffer[Tree]
       while (in.token == AT) {
         in.nextToken()
-        annotation()
+        annotation() match {
+          case Some(anno) => annots += anno
+          case _ =>
+        }
       }
-      List() // don't pass on annotations for now
+      annots.toList
     }
 
     /** Annotation ::= TypeName [`(` AnnotationArgument {`,` AnnotationArgument} `)`]
       */
-    def annotation(): Unit = {
-      qualId()
-      if (in.token == LPAREN) { skipAhead(); accept(RPAREN) }
-      else if (in.token == LBRACE) { skipAhead(); accept(RBRACE) }
+    def annotation(): Option[Tree] = {
+      val id = convertToTypeId(qualId())
+      var skipAnno = false
+
+      // only parse annotations without arguments
+      if (in.token == LPAREN) {
+        in.nextToken()
```

Comment:
Instead of calling `nextToken()` which then requires you to call `skipAhead` with special arguments, I think you could call `in.lookaheadToken` to directly check if the next token is an RPAREN.

Reply:
the change to `skipAhead` is reverted